app/index.js

```diff
@@ -1,214 +1,233 @@
-
-			// Copy AngularJS files
-			this.directory('public/modules/articles');
-		}
-	},
-
-	renderApplicationEnvironmentConfigFiles: function() {
-		this.template('config/env/_all.js', 'config/env/all.js');
-		this.template('config/env/_development.js', 'config/env/development.js');
-		this.template('config/env/_production.js', 'config/env/production.js');
-		this.template('config/env/_test.js', 'config/env/test.js');
-		this.template('config/env/_secure.js', 'config/env/secure.js');
-	},
-
-	renderAngularApplicationConfigFile: function() {
-		this.template('public/_config.js', 'public/config.js');
-	},
-
-	renderCoreModuleFiles: function() {
-		this.template('public/modules/core/views/_header.client.view.html', 'public/modules/core/views/header.client.view.html');
-	},
-
-	renderApplicationDependenciesFiles: function() {
-		this.template('_package.json', 'package.json');
-		this.template('_bower.json', 'bower.json');
-	}
+  
+  init: function () {
+    this.pkg = yeoman.file.readJSON(path.join(__dirname, '../package.json'));
+
+    this.on('end', function () {
+      clog('');
+      clog.green('------------------------------------------');
+      clog.green('Your MEAN.js application is ready!');
+      clog('');
+      clog.green('To Get Started, run the following command:');
+      clog('');
+      clog.blue('cd ' + folder + ' && npm install && grunt');
+      clog('');
+      clog.green('Happy Hacking!');
+      clog.green('------------------------------------------');
+    });
+  },
+
+  checkForGit: function () {
+    var done = this.async();
+
+    exec('git --version')
+      .then(function () {
+        done();
+      })
+      .catch(function (err) {
+        clog.red(new Error(err));
+        return;
+      });
+  },
+
+  welcomeMessage: function () {
+    clog(this.yeoman);
+
+    clog.green('You\'re using the official MEAN.JS generator.');
+  },
+
+  promptForVersion: function () {
+    var done = this.async();
+
+    var choices = [];
+    for(var ver in versions) {
+      choices.push(ver);
+    }
+
+    var prompt = {
+      type: 'list',
+      name: 'version',
+      message: 'What mean.js version would you like to generate?',
+      choices: choices,
+      default: 1
+    };
+
+    this.prompt(prompt, function (props) {
+      version = props.version;
+      done();
+    }.bind(this));
+
+  },
+
+  promptForFolder: function () {
+    var done = this.async();
+
+    clog.red(version);
+
+    var prompt = {
+      name: 'folder',
+      message: 'In which folder would you like the project to be generated? This can be changed later.',
+      default: 'mean'
+    };
+
+    this.prompt(prompt, function (props) {
+      folder = props.folder;
+      folderPath = './' + folder + '/';
+      done();
+    }.bind(this));
+  },
+
+  cloneRepo: function () {
+    var done = this.async();
+
+    exec('git clone --branch ' + versions[version] + ' https://github.com/meanjs/mean.git ' + folder)
+      .then(function () {
```

Comment:
There could be instances where we need to use it. Like when git --version res would be the version of git. Currently, it's not being used though.